### PR TITLE
CI: replace eslint and prettier with biome

### PR DESCRIPTION
**Pull request by soofstad**
Replace eslint and prettier with biome

Why? Because Biome works, eslint does not

### CODE REVIEW

**Comment by sebastianvitterso** [comment]
> Because Biome works, eslint does not

SHOTS FIRED

File: examples/microsoft-auth-provider/web-app/src/index.tsx [hunk 1]
```diff
@@ -26,7 +26,9 @@ function LoginInfo(): JSX.Element {
     return (
       <>
         <div style={{ color: 'red' }}>An error occurred during authentication: {error}</div>
-        <button onClick={() => logOut()}>Logout</button>
+        <button type={'button'} onClick={() => logOut()}>
```
Comment (sebastianvitterso) [inline examples/microsoft-auth-provider/web-app/src/index.tsx:29]: Surely `type='button'` is simpler than `type={'button'}`?
```suggestion
        <button type='button' onClick={() => logOut()}>
```

File: src/AuthContext.tsx [hunk 2]
```diff
@@ -117,10 +115,20 @@ export const AuthProvider = ({ authConfig, children }: IAuthProvider) => {
 
   function handleExpiredRefreshToken(initial = false): void {
     // If it's the first page load, OR there is no sessionExpire callback, we trigger a new login
-    if (initial) return login()
+    if (initial) {
+      login()
+      return
+    }
```
Comment (sebastianvitterso) [inline src/AuthContext.tsx:121]: Ah, I too would have preferred `return login()`, but this is equally as understandable.

File: src/Types.ts [hunk 3]
```diff
@@ -19,6 +19,7 @@ export interface TTokenRequestForRefresh extends TTokenRqBase {
 export type TTokenRequest = TTokenRequestWithCodeAndVerifier | TTokenRequestForRefresh
 
 export type TTokenData = {
+  // biome-ignore lint: reason
   [x: string]: any
 }
```
Comment (sebastianvitterso) [inline src/Types.ts:24]: Didn't it allow the `any`?

Btw, I think the "reason" is something you can replace with your reason for the ignore 😅

File: src/authentication.ts [hunk 4]
```diff
@@ -120,7 +119,7 @@ export function redirectToLogout(
     token_type_hint: refresh_token ? 'refresh_token' : 'access_token',
     client_id: config.clientId,
     post_logout_redirect_uri: config.logoutRedirect ?? config.redirectUri,
-    ui_locales: window.navigator.languages.reduce((a: string, b: string) => a + ' ' + b),
+    ui_locales: window.navigator.languages.reduce((a: string, b: string) => `${a} ${b}`),
```
Comment (sebastianvitterso) [inline src/authentication.ts:122]: Never seen a `.reduce` like this 😅  Why not just do this?
```suggestion
    ui_locales: window.navigator.languages.join(' '),
```

**Review by sebastianvitterso — DISMISSED** [review]
Looks pretty good, but a couple of notes!

File: src/authentication.ts (hunk 4 above)
Comment (soofstad) [reply src/authentication.ts:122]: Yeah.... no idea why I did that

**Pull request merged**